### PR TITLE
Egonet Implementation for Jasminegraph

### PR DESCRIPTION
**Pull request by kasundharmadasa**

### CODE REVIEW

File: src/frontend/JasmineGraphFrontEnd.cpp [hunk 1]
```diff
@@ -1067,6 +1067,50 @@ void *frontendservicesesion(std::string masterIP, int connFd, SQLiteDBInterface
...
+            result_wr = write(connFd, "\r\n", 2);
+            if (result_wr < 0) {
+                frontend_logger.log("Error writing to socket", "error");
+                loop = true;
+                continue;
+            }
+
+            char graph_id[FRONTEND_DATA_LENGTH];
+            bzero(graph_id, FRONTEND_DATA_LENGTH + 1);
+            string name = "";
```
Comment (miyurud) [inline src/frontend/JasmineGraphFrontEnd.cpp:1100]: This variable seems not used at all.

File: src/frontend/JasmineGraphFrontEnd.cpp [hunk 2]
```diff
@@ -1067,6 +1067,50 @@ void *frontendservicesesion(std::string masterIP, int connFd, SQLiteDBInterface
...
+            string path = "";
```
Comment (miyurud) [inline src/frontend/JasmineGraphFrontEnd.cpp:1101]: The variable `path` seems not used at all.

File: src/server/JasmineGraphInstanceService.cpp [hunk 3]
```diff
@@ -614,62 +614,54 @@ void *instanceservicesession(void *dummyPt) {
...
+            map<long, long> degreeDistribution = calculateLocalInDegreeDist(graphID, partitionID,
+                                                                            graphDBMapLocalStores,
+                                                                            graphDBMapCentralStores);
+            instance_logger.log("In Degree Dist size: " + to_string(degreeDistribution.size()), "info");
 
-            write(connFd, JasmineGraphInstanceService::END_OF_MESSAGE.c_str(),
-                  JasmineGraphInstanceService::END_OF_MESSAGE.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceService::END_OF_MESSAGE, "info");
-        } else if (line.compare(JasmineGraphInstanceProtocol::DP_CENTRALSTORE) == 0) {
+            //todo kasundharmadasa : The in degree distribution of a worker is kept in the worker itself
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:660]: Do we keep the in-degree distribution of each partition in memory? or do we store them in secondary storage?

File: src/server/JasmineGraphInstanceService.cpp [hunk 4]
```diff
@@ -741,185 +767,127 @@ void *instanceservicesession(void *dummyPt) {
...
-                 ++workerIt) {
-                std::vector<string> workerSocketPair;
-                stringstream wl(*workerIt);
-                string intermediate;
-                while (getline(wl, intermediate, ':')) {
-                    workerSocketPair.push_back(intermediate);
-                }
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string workerList = (data);
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:801]: Can't we just change this line as follows,

`string workerList = data;`

**Review by miyurud — CHANGES_REQUESTED** [review]

File: src/frontend/JasmineGraphFrontEnd.cpp (hunk 1 above)
Comment (kasundharmadasa) [reply src/frontend/JasmineGraphFrontEnd.cpp]: Resolved

File: src/frontend/JasmineGraphFrontEnd.cpp (hunk 2 above)
Comment (kasundharmadasa) [reply src/frontend/JasmineGraphFrontEnd.cpp]: resolved

File: src/server/JasmineGraphInstanceService.cpp (hunk 4 above)
Comment (kasundharmadasa) [reply src/server/JasmineGraphInstanceService.cpp]: resolved

File: src/frontend/JasmineGraphFrontEnd.cpp (hunk 1 above)
Comment (miyurud) [reply src/frontend/JasmineGraphFrontEnd.cpp]: Ok

File: src/frontend/JasmineGraphFrontEnd.cpp (hunk 2 above)
Comment (miyurud) [reply src/frontend/JasmineGraphFrontEnd.cpp]: Ok

File: src/server/JasmineGraphInstanceService.cpp (hunk 3 above)
Comment (miyurud) [reply src/server/JasmineGraphInstanceService.cpp:660]: Ok

File: src/server/JasmineGraphInstanceService.cpp (hunk 4 above)
Comment (miyurud) [reply src/server/JasmineGraphInstanceService.cpp]: Ok

**Review by miyurud — APPROVED** [review]

**Pull request merged**